sql/sql_audit.cc

```diff
@@ -384,7 +388,41 @@ int mysql_audit_notify(THD *thd, mysql_event_general_subclass_t subclass,
   event.general_host = sctx->host();
   event.general_external_user = sctx->external_user();
   event.general_rows = thd->get_stmt_da()->current_row_for_condition();
-  event.general_sql_command = sql_statement_names[thd->lex->sql_command];
+  if (thd->lex->sql_command == SQLCOM_END && msg_len > 0 && error_code == 0) {
+    int found_index = -1;
+    for (size_t i = 0; i < get_command_name_len(); ++i) {
+      if (!strcmp(command_name[i].str, msg)) {
+        found_index = i;
+        break;
+      }
+    }
+
+    if (found_index == -1) {
+      event.general_sql_command = sql_statement_names[thd->lex->sql_command];
+    } else {
+      switch (found_index) {
+        case COM_STMT_PREPARE:
+          event.general_sql_command = sql_statement_names[SQLCOM_PREPARE];
+          break;
+        case COM_STMT_EXECUTE:
+          event.general_sql_command = sql_statement_names[SQLCOM_EXECUTE];
+          break;
+        case COM_STMT_RESET:
+          event.general_sql_command = sql_statement_names[SQLCOM_RESET];
+          break;
+        default:
+          cmd_class_lowercase = msg;
+          boost::to_lower(cmd_class_lowercase);
```

Comment:
like user_buf object on stack here. why can't we use std::string local to function?

Reply:
Just placing std::string at start of function is sufficient? please check

Reply:
Also would prefer std library compared to boost (when possible)

Reply:
Check how user_buff access via event.general_user.str is safe( or not safe?)

Reply:
It's safe , Since logging to audit_log happens before cmd_class_lowercase goes out of scope.